### ISSUE TITLE
[1.12.2] Possible conflict with 'Scaling Health'. Damage gets scaled quite a bit if blocked with a Tinker's Battlesign while having 'Spiky' if you have Defender Core 4 unlocked.

### ISSUE THREAD

**Issue by SonicX8000** [Tamaized/AoV#91]
Refer'd Issue:
https://github.com/SilentChaos512/ScalingHealth/issues/146

Minecraft: 1.12.2
Forge: 14.23.5.2795

Mods: 6

AoV-1.12-0.9.4
Mantle-1.12-1.3.3.39
TamModized-1.12.2-0.15.6
TConstruct-1.12.2-2.11.0.106
ScalingHealth-1.12.2-1.3.32-134
SilentLib-1.12.2-3.0.9-159

-----

Scaling Health has an option of scaling up the damage taken from damage sources depending on how you have it config'd. For example... you can have cacti deal 1000000 damage if you really wanted with pretty much makes anything that deals the 'cactus' damage source. Anyways...

Copied from a discord discussion...

*So... I think I ran into a possible conflict with Angel of Vengeance, there's a skill in that mod that allows Shields to have a sort of 'round' block while guarding. When combined with a Battlesign with the 'Spiky' trait from Tinker's... the damage gets scaled and then reflected back at the mob dealing the damage which can deal MASSIVE damage.*

*Here's the value without AoV's Defender Core 4 Skill, which is the 'Round Block' and having a Zombie Pigman attack me while blocking with a 'Spiky' Battlesign.*

>Damage scaling: type=mob, scale=2.0, affected=125.0, change=0.0, original=0.0

*The damage the Zombie Pigman takes is 51, due to the stats on the battlesign.*

*Now's here's the value with AoV's Defender Core 4 Skill, and having a Zombie Pigman attack me while blocking with a 'Spiky' Battlesign*

>Damage scaling: type=mob, scale=2.0, affected=125.0, change=2000.0, original=8.0

*The damage the Zombie Pigman takes is 430, due to the scaled damage being reflected as well as the stats on the Battlesign.*

*Not sure if it's on AoV's side or Scaling Health's side. I am certain that the 'Affect Hostile/Passive Mobs' is unchecked. It could be a missing check or maybe the way that the priority is set. It might be fixed already in your next Scaling Health version since you did a change in your damage scaling code related to the DE Energy Shield issue that was mentioned awhile ago.*

-----

It only happens when you block with a Battlesign with the 'Spiky' trait on either the sign itself or a weapon. If I use any other shield alongside a 'Spiky' weapon... the reflected damage isn't scaled and deals the normal amount of damage.

I wonder if this is a side effect of having the 'Spiky' trait work with the Defender Core 4 skill. Figure I give a post here just in case.

**Comment by Tamaized** [comment]
I'm having trouble reproducing this. Can you give me the EXACT setup to get these results, configs and all as well since I've never used Scaling Health before.

**Comment by SonicX8000** [comment]
I'll give you my config settings which should have the results. You will need to drop this file into the into 'config > scalinghealth' folder, replacing the 'main.cfg' file that's currently in there.

File:
https://cdn.discordapp.com/attachments/377171551155650560/535149658146865156/Scaling_Health_Config.zip

To increase the difficulty I think the command is....

*/scalinghealth difficulty set "value"*

In the config above... the max value for difficulty is 500 so only 0 to 500 for "value" will work. To set health you do...

*/scalinghealth health set "value"*

The player max health cap in the config is 0, meaning it's uncapped... to a point. There's a 'Max health Cap' config option under all the options if you check it's in-game config. That value is a overall cap for both players & mobs. I did 2000 during the test, set it higher if you wish.

Now the steps that I do. "keepInventory = true" will be handy since you'll perhaps you'll get one-shot.

I have a Manyullyn Battlesign with the 'Magma Slime Sign Plate' & 'Cactus Tool Rod', this gives the tool traits 'Superheat' which does bonus damage vs enemies on fire and 'Spiky' which reflects damage back to the attacker. Any sign plate works but you only need the 'Spiky' trait.

Due to the high damage because of scaling health's difficulty & config settings, the Battlesign will instantly break if it blocks an attack. To make testing easier add 2 Creative Modifiers or level up the tool 2 times or so... whichever works so you can add 5 Reinforcment Plates to make the sign 'Unbreakable'.

Now get hit by a mob, any mob whether it be an arrow or a melee attack WITHOUT 'Defender Core 4' unlocked. You'll see that the mob takes some damage but not a lot. Now do the same WITH 'Defender Core 4' unlocked. The mob will take heavy damage... so heavy that if double strike triggers you'll pretty much outright 1-shot the mob... maybe.

if you want to see damage values... you can install the 'ToroHud Damage Indicators' mod which shows a damage particle for mobs or players that take damage. It also shows healing particles.

To add onto this... while doing a test, ToroHud shows the blocked hit and something else... I even managed to have double strike occur. Zombie Pigman had about 1388-1389 health and in-gmae difficulty was Hard with 500 scalinghealth difficulty, the difficulty weight was at 2.0 and Scaling Mode set to 'PLAYER_DIFFICULTY'.

![](https://cdn.discordapp.com/attachments/377171551155650560/535150557879336961/battlesignouch.PNG)
Left side:
The 13 is Spiky reflecting the damage due to the hit being blocked, scaling health doesn't scale this damage so it's the original value.

Right side:
The 13 is Spiky reflecting the damage due to the hit being blocked, however with Defender Core 4 apparently another reflected hit occurs which is the 554. Double Strike also proc'd for that value which is why there's two of them. That's a whopping 1108 damage reflected back. Totaling to 1121 damage if adding the 13 onto that.

**Comment by Tamaized** [comment]
Holy fucking shit that was a massive headache to debug. EXTREMELY simple fix though.

Turns out it was a conflict between just AoV and Tinkers alone. I was accidentally triggering tinker's spikey a second time with an incorrect value.

**Comment by SonicX8000** [comment]
Ah, I thought it was scaling health at first due to the reflected damage being scaled while blocking... until I saw that the damage is shown separated via Toro HUD. For damage particles when I first discovered it I was using dynamic surrounding's damage particles and those add up the damage in total while ToroHUD shows them individually.

*EDIT*
Bug is fixed in "AoV-1.12-0.9.5". Damage is only reflected once instead of twice without the massive damage.
